src/main/java/seedu/address/ui/CalendarPanel.java

```diff
@@ -0,0 +1,118 @@
+package seedu.address.ui;
+
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.YearMonth;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Region;
+
+import seedu.address.model.events.EventSource;
+
+/**
+ * An Ui that stores the logged feedback from the program to the user.
+ */
+public class CalendarPanel extends UiPart<Region> {
+
+    private static final String FXML = "CalendarPanel.fxml";
+
+    private YearMonth yearMonth;
+    private LocalDate calendarDate;
+
+    @FXML
+    private GridPane calendar;
+
+    @FXML
+    private Label month;
+
+    /**
+     * Constructor for ListPanel. Stores the event list, and task list[in v2.0].
+     */
+    public CalendarPanel(ObservableList<EventSource> eventList) {
+        super(FXML);
+        this.yearMonth = YearMonth.now();
+        this.calendarDate = LocalDate.now();
+        setCurrentMonth();
+        fillIndexOfCalendar(calendarDate.getDayOfWeek().getValue(), eventList);
+
+    }
+
+    /**
+     * Sets the label to be the current Month.
+     */
+    private void setCurrentMonth() {
+        String currentMonth = yearMonth.getMonth().toString().toLowerCase();
+        currentMonth = currentMonth.substring(0, 1).toUpperCase() + currentMonth.substring(1);
+        month.setText(currentMonth);
+    }
+
+    /**
+     * Fills the index of the calendar.
+     * @param startingDay Indicates the starting day of the week
+     * @param eventList Contains the EventSource
+     */
+    private void fillIndexOfCalendar(int startingDay, ObservableList<EventSource> eventList) {
+        int index = 1;
+        int totalDays = yearMonth.lengthOfMonth();
+        for (int i = 0; i < 6; i++) {
```

Comment:
Could probably rename the counter variables with more meaningful names and/or abstract out this block into another function.